blobstore-fs/src/main.rs

```diff
@@ -0,0 +1,676 @@
+
+        let root_val = match values.get("root") {
+            None => "/tmp",
+            Some(r) => r.as_str(),
+        };
+    
+        let config = FsProviderConfig {
+            ld: ld.clone(),
+            root: PathBuf::from(root_val),
+        };
+
+        info!("Config: {:?}", config);
+
+        info!("File System Blob Store Container Root: '{:?}'", &config.root);
+
+        self.config
+            .write()
+            .await
+            .insert(ld.actor_id.clone(), config.clone());
+
+        // Create a directory named from the actor id:
+        let cdir = Path::new(&config.root).join(Path::new(&ld.actor_id));
+
+        match std::fs::create_dir_all(cdir.as_path()) {
+            Ok(()) => Ok(true),
+            Err(e) => Err(RpcError::InvalidParameter(format!(
+                "Could not create actor directory: {:?}",
+                e
+            ))),
+        }
+
+    }
+}
+
+/// Handle Factorial methods
+#[async_trait]
+impl Blobstore for FsProvider {
+    /// Returns whether the container exists
+    #[allow(unused)]
+    async fn container_exists(&self, ctx: &Context, arg: &ContainerId) -> RpcResult<bool> {
+        info!("Called container_exists({:?})", arg);
+
+        let root = self.get_root(ctx).await?;
+        let cdir = Path::new(&root).join(&arg);
+
+        match read_dir(&cdir) {
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
+    }
+
+    /// Creates a container by name, returning success if it worked
+    /// Note that container names may not be globally unique - just unique within the
+    /// "namespace" of the connecting actor and linkdef
+    async fn create_container(&self, ctx: &Context, arg: &ContainerId) -> RpcResult<()> {
+        let root = self.get_root(ctx).await?;
+        let cdir = Path::new(&root).join(arg.to_string());
+
+        info!("create dir: {:?}", cdir);
+
+        match std::fs::create_dir_all(cdir) {
+            Ok(()) => Ok(()),
+            Err(e) => Err(RpcError::InvalidParameter(format!(
+                "Could not create container: {:?}",
+                e
+            ))),
+        }
+    }
+
+    /// Retrieves information about the container.
+    /// Returns error if the container id is invalid or not found.
+    #[allow(unused)]
+    async fn get_container_info(
+        &self,
+        ctx: &Context,
+        arg: &ContainerId,
+    ) -> RpcResult<ContainerMetadata> {
+        let root = self.get_root(ctx).await?;
+        let dir_path = Path::new(&root).join(&arg);
+
+        let dir_info = metadata(dir_path)?;
+
+        let modified = match dir_info.modified()?.duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(s) => Timestamp {
+                sec: s.as_secs() as i64,
+                nsec: 0u32,
+            },
+            Err(e) => return Err(RpcError::InvalidParameter(format!("{:?}", e))),
+        };
+
+        Ok(ContainerMetadata {
+            container_id: arg.clone(),
+            created_at: Some(modified),
```

Comment:
I don't really know why actors need to be aware of the creation time of a particular container, but that's not an issue for this PR, just something to consider for future revisions to the blobstore contract.

Reply:
I did not really ponder on this, but it is a consequence of the interface proposed by @stevelr